source/core/module_coreHelpers.js

```diff
@@ -75,33 +104,68 @@
                     str += String.fromCharCode(u0);
                     continue;
                 }
+
+                // Look ahead to validate the UTF-8 structure
+                if ((u0 & 0xE0) === 0xC0) {
```

Comment:
Seeing these hex values makes it hard to understand. At first I thought adding some comments, but perhaps declaring constants could help too.